liquid-rust-typeck/src/ty.rs

```diff
@@ -395,6 +404,28 @@ impl TyS {
         matches!(self.kind(), TyKind::Uninit(..))
     }
 
+    pub fn vars_worker(&self, vars: &mut FxHashSet<Name>) {
```

Comment:
Maybe `gather_vars`? Also, let's make this private, I think it's only being used in this file.
```suggestion
    fn gather_vars(&self, vars: &mut FxHashSet<Name>) {
```